# Patch against libcharset version 1.1
--- /home/jas/src/libiconv/libcharset//lib/localcharset.c	2002-05-10 12:40:10.000000000 +0200
+++ localcharset.c	2002-10-07 00:15:05.000000000 +0200
@@ -102,8 +102,8 @@
 static const char * volatile charset_aliases;
 
 /* Return a pointer to the contents of the charset.alias file.  */
-static const char *
-get_charset_aliases ()
+const char *
+_gsasl_locale_get_charset_aliases ()
 {
   const char *cp;
 
@@ -239,7 +239,7 @@
 STATIC
 #endif
 const char *
-locale_charset ()
+_gsasl_locale_charset ()
 {
   const char *codeset;
   const char *aliases;
@@ -349,7 +349,7 @@
     codeset = "";
 
   /* Resolve alias. */
-  for (aliases = get_charset_aliases ();
+  for (aliases = _gsasl_locale_get_charset_aliases ();
        *aliases != '\0';
        aliases += strlen (aliases) + 1, aliases += strlen (aliases) + 1)
     if (strcmp (codeset, aliases) == 0
--- /home/jas/src/libiconv/libcharset//include/libcharset.h.in	2001-03-27 15:35:32.000000000 +0200
+++ libcharset.h	2002-10-07 00:15:05.000000000 +0200
@@ -30,8 +30,8 @@
    The result must not be freed; it is statically allocated.
    If the canonical name cannot be determined, the result is a non-canonical
    name.  */
-extern const char * locale_charset (void);
-
+extern const char * _gsasl_locale_charset (void);
+extern const char * _gsasl_locale_get_charset_aliases (void);
 
 #ifdef __cplusplus
 }
