--- gl/getpass.c.orig	2010-01-12 12:48:52.000000000 +0100
+++ gl/getpass.c	2010-01-12 12:51:39.000000000 +0100
@@ -192,6 +192,7 @@
   char getpassbuf[PASS_MAX + 1];
   size_t i = 0;
   int c;
+  int use_getc = 0;
 
   if (prompt)
     {
@@ -201,8 +202,20 @@
 
   for (;;)
     {
-      c = _getch ();
-      if (c == '\r')
+      if (use_getc)
+        c = getc (stdin);
+      else
+        {
+          c = _getch ();
+          if (c == -1)
+            {
+              /* Under Wine, _getch always returns -1.  Fall back to
+                 using getc.  */
+              use_getc = 1;
+              continue;
+            }
+        }
+      if (c == '\n' || c == '\r')
         {
           getpassbuf[i] = '\0';
           break;
@@ -219,7 +232,7 @@
         }
     }
 
-  if (prompt)
+  if (prompt && !use_getc)
     {
       fputs ("\r\n", stderr);
       fflush (stderr);
