--- gendocs_template.orig	2008-07-01 15:40:29.000000000 +0200
+++ gendocs_template	2008-07-01 15:40:42.000000000 +0200
@@ -59,6 +59,19 @@
     (%%TEXI_TGZ_SIZE%%K bytes gzipped tar file).</a></li>
 </ul>
 
+<p>There is also an API reference manual:</p>
+
+<ul>
+  <li><a href="../doxygen/">Doxygen HTML</a>.</li>
+  <li><a href="../doxygen/%%PACKAGE%%.pdf">Doxygen PDF file</a>.</li>
+</ul>
+
+<ul>
+  <li><a href="../reference/">HTML API Reference Manual (GTK-DOC)</a>.</li>
+  <li><a href="../reference/%%PACKAGE%%.pdf">PDF Reference API Manual (GTK-DOC)</a>.
+  <li><a href="../reference/gsasl.devhelp">DevHelp API Reference Manual (GTK-DOC)</a>.</li>
+</ul>
+
 <p>You can <a href="http://shop.fsf.org/">buy printed copies of
 some manuals</a> (among other items) from the Free Software Foundation;
 this helps support FSF activities.</p>
